### PR TITLE
Added stylesheet file setting

### PR DESCRIPTION
**Pull request by sigzegv**
I just wanted to show this update that allows to use a custom stylesheet in config file or autocommands. I'm not sure if this is the correct way, and I'm not sure about naming conventions. Tell me about any needed modifications.

With that I wonder if the stylesheet boolean parameter is still necessary, because having a definder stylesheet file or not makes  same "boolean" purpose.

### CODE REVIEW

**Comment by sigzegv** [comment]
the link hinting doesn't work with this code, don't really found why yet.. Any ideas welcomed.

File: src/setting.c [hunk 1]
```diff
@@ -693,6 +696,32 @@ static int user_style(Client *c, const char *name, DataType type, void *value, v
     return CMD_SUCCESS;
 }
 
+static int style_sheet_file(Client *c, const char *name, DataType type, void *value, void *data)
+{
+    char *path;
+    gchar *source;
+
+    path = util_get_config_dir();
```
Comment (fanglingsu) [inline src/setting.c]: This variable has to be g_freed after usage.

File: src/setting.c [hunk 2]
```diff
@@ -693,6 +696,32 @@ static int user_style(Client *c, const char *name, DataType type, void *value, v
...
+    ucm = webkit_web_view_get_user_content_manager(c->webview);
+
+    WebKitUserStyleSheet *style;
+    g_file_get_contents(vb.files[FILES_USER_STYLE], &source, NULL, NULL);
+    style = webkit_user_style_sheet_new(
+        source, WEBKIT_USER_CONTENT_INJECT_ALL_FRAMES,
+        WEBKIT_USER_STYLE_LEVEL_USER, NULL, NULL
+    );
+
+    webkit_user_content_manager_add_style_sheet(ucm, style);
```
Comment (fanglingsu) [inline src/setting.c]: I think you have to remove previouly set style sheets. Else the styles will be merged together internally.

File: src/setting.c [hunk 3]
```diff
@@ -693,6 +696,32 @@ static int user_style(Client *c, const char *name, DataType type, void *value, v
...
 }
 
+static int style_sheet_file(Client *c, const char *name, DataType type, void *value, void *data)
+{
+    char *path;
+    gchar *source;
+
+    path = util_get_config_dir();
+
+    vb.files[FILES_USER_STYLE] = util_get_filepath(path, value, FALSE);
```
Comment (fanglingsu) [inline src/setting.c]: Note that this contains already allocated memory and has to be freed before assigning new.

**Comment by fanglingsu** [comment]
> the link hinting doesn't work with this code, don't really found why yet.. Any ideas welcomed.

This is because the logic to add the hinting style sheet is added in `user_style` function to make sure the style sheet is added again after changeing the user style sheet.

**Pull request closed without merging**